quic/s2n-quic-core/src/transport/parameters/mod.rs

```diff
@@ -373,7 +381,8 @@ macro_rules! duration_transport_parameter {
     };
 }
 
-/// Implements an optional transport parameter
+/// Implements an optional transport parameter. This is used to define transport
+/// parameters that are only sent by one peer in a connection.
```

Comment:
The single side parameters are generics and use the Disabled type. The optional values are for parameters that don't really have a good default. For example, an IP address needs to be provided for the server preferred address. It's not really possible to fill that value in to something.

Reply:
I'm going to add that as a comment, its nice to know what these macros are for.